cylc/flow/graph_parser.py

```diff
@@ -85,98 +85,110 @@ class GraphParser:
     OP_OR_ERR = '||'
     SUICIDE = '!'
     OPTIONAL = '?'
-    TRIGGER = ':'
+    QUALIFIER = ':'
     ARROW = '=>'
-    ACTION = '@'
-
-    TRIG_FAM_SUCCEED_ALL = "succeed-all"
-    TRIG_FAM_SUCCEED_ANY = "succeed-any"
-    TRIG_FAM_FAIL_ALL = "fail-all"
-    TRIG_FAM_FAIL_ANY = "fail-any"
-    TRIG_FAM_FINISH_ALL = "finish-all"
-    TRIG_FAM_FINISH_ANY = "finish-any"
-    TRIG_FAM_START_ALL = "start-all"
-    TRIG_FAM_START_ANY = "start-any"
-    TRIG_FAM_SUBMIT_ALL = "submit-all"
-    TRIG_FAM_SUBMIT_ANY = "submit-any"
-    TRIG_FAM_SUBMIT_FAIL_ALL = "submit-fail-all"
-    TRIG_FAM_SUBMIT_FAIL_ANY = "submit-fail-any"
-
-    # Map family trigger to inferred member output optionality.
-    fam_output_map = {
-        # Required outputs.
-        # (started is never optional; only checked if task finishes)
-        TRIG_FAM_START_ALL: (TASK_OUTPUT_STARTED, False),
-        TRIG_FAM_START_ANY: (TASK_OUTPUT_STARTED, False),
-        TRIG_FAM_SUCCEED_ALL: (TASK_OUTPUT_SUCCEEDED, False),
-        TRIG_FAM_FAIL_ALL: (TASK_OUTPUT_FAILED, False),
-        TRIG_FAM_SUBMIT_ALL: (TASK_OUTPUT_SUBMITTED, False),
-        TRIG_FAM_SUBMIT_FAIL_ALL: (TASK_OUTPUT_SUBMIT_FAILED, False),
-        # Optional outputs.
-        TRIG_FAM_SUBMIT_FAIL_ANY: (TASK_OUTPUT_SUBMITTED, True),
-        TRIG_FAM_SUCCEED_ANY: (TASK_OUTPUT_SUCCEEDED, True),
-        TRIG_FAM_FAIL_ANY: (TASK_OUTPUT_SUCCEEDED, True),
-        TRIG_FAM_FINISH_ALL: (TASK_OUTPUT_SUCCEEDED, True),
-        TRIG_FAM_FINISH_ANY: (TASK_OUTPUT_SUCCEEDED, True),
-        TRIG_FAM_SUBMIT_ANY: (TASK_OUTPUT_SUBMITTED, True),
+    XTRIG = '@'
+
+    QUAL_FAM_SUCCEED_ALL = "succeed-all"
+    QUAL_FAM_SUCCEED_ANY = "succeed-any"
+    QUAL_FAM_FAIL_ALL = "fail-all"
+    QUAL_FAM_FAIL_ANY = "fail-any"
+    QUAL_FAM_FINISH_ALL = "finish-all"
+    QUAL_FAM_FINISH_ANY = "finish-any"
+    QUAL_FAM_START_ALL = "start-all"
+    QUAL_FAM_START_ANY = "start-any"
+    QUAL_FAM_SUBMIT_ALL = "submit-all"
+    QUAL_FAM_SUBMIT_ANY = "submit-any"
+    QUAL_FAM_SUBMIT_FAIL_ALL = "submit-fail-all"
+    QUAL_FAM_SUBMIT_FAIL_ANY = "submit-fail-any"
+
+    # Map family trigger type to (member-trigger, any/all), for use in
+    # expanding family trigger expressions to member trigger expressions.
+    # - "FAM:succeed-all => g" means "f1:succeed & f2:succeed => g"
+    # - "FAM:fail-any => g" means "f1:fail | f2:fail => g".
+    # E.g. QUAL_FAM_START_ALL: (TASK_OUTPUT_STARTED, True) simply maps
+    #   "FAM:start-all" to "MEMBER:started" and "-all" (all members).
+    fam_to_mem_trigger_map = {
+        QUAL_FAM_START_ALL: (TASK_OUTPUT_STARTED, True),
+        QUAL_FAM_START_ANY: (TASK_OUTPUT_STARTED, False),
```

Comment:
Thanks for clarifying this, but I'm still not sure what the boolean means here. Is it literally just `is_all`?

Reply:
Correct. The alternative to use `.endswith('-all')` on the family qualifier, but I figured a static map is slightly more efficient.